All_home_works/hw1/task4.py

```diff
@@ -0,0 +1,30 @@
+"""
+Given a list of integers numbers "nums".
+You need to find a sub-array with length less equal to "k", with maximal sum.
+The written function should return the sum of this sub-array.
+Examples:
+    nums = [1, 3, -1, -3, 5, 3, 6, 7], k = 3
+    result = 16
+"""
+from typing import List
+
+
+def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
+    counter = 0
+    max_sum = 0
+
+    for index, current_num in enumerate(nums):
+        if index < k:
+            counter += current_num
+            max_sum += current_num
+        else:
+            num_k_indexes_ago = nums[index - k]
+            counter += current_num - num_k_indexes_ago
+            max_sum = _get_max_sum(counter, max_sum)
+    return max_sum
+
+
+def _get_max_sum(counter, max_sum):
```

Comment:
This function looks familiar, is anything similar in the standard library?

Reply:
In the standard library, we have both: sum() and max(). Maybe it's not a very good naming of my function, and I do not see a lot of sense do it as a function, but flake8 has a problem with cognitive complexity. It's 7 or 8 if I do not separate it into functions.

Reply:
Now, this function is another. Sorry, but I still do not understand it.

Reply:
The old version could be replaced with `max`. You just implemented something that already present in the standard library.